pkg/locodedb/calls.go

```diff
@@ -24,40 +24,62 @@ func Get(locodeStr string) (Record, error) {
 		return Record{}, err
 	}
 
-	country := locodeStr[:2]
-	location := locodeStr[2:]
-
-	if locodeStr[2] == ' ' {
-		location = locodeStr[3:]
+	if len(locodeStr) == CountryCodeLen+LocationCodeLen+1 && locodeStr[CountryCodeLen] == ' ' {
+		locodeStr = locodeStr[:CountryCodeLen] + locodeStr[CountryCodeLen+1:]
 	}
-
-	key, err := NewKey(country, location)
-	if err != nil {
-		return Record{}, err
+	if len(locodeStr) != CountryCodeLen+LocationCodeLen {
+		return Record{}, ErrInvalidString
 	}
-	return getFromKey(*key)
-}
 
-// getFromKey returns a record for a given Key.
-func getFromKey(key Key) (Record, error) {
-	newlocode := key.CountryCode().String() + key.LocationCode().String()
+	for i := range locodeStr[:CountryCodeLen] {
+		if !isUpperAlpha(locodeStr[i]) {
+			return Record{}, ErrInvalidString
+		}
+	}
+	for i := range locodeStr[CountryCodeLen:] {
+		if !isUpperAlpha(locodeStr[CountryCodeLen+i]) && !isDigit(locodeStr[CountryCodeLen+i]) {
+			return Record{}, ErrInvalidString
+		}
+	}
 
-	locodeCSV, found := mLocodes[newlocode]
-	if !found {
+	cc := CountryCode{}
+	copy(cc[:], locodeStr[:2])
+	cd, countryFound := mCountries[cc]
+	if !countryFound {
 		return Record{}, ErrNotFound
 	}
 
-	country, countryFound := mCountries[*key.CountryCode()]
-	if !countryFound {
+	code := locodeStr[CountryCodeLen:]
+	n := sort.Search(len(cd.locodes), func(i int) bool {
+		cmp := strings.Compare(codeFromCSV(&cd.locodes[i]), code)
+		return cmp >= 0
+	})
+	if n == len(cd.locodes) || strings.Compare(codeFromCSV(&cd.locodes[n]), code) != 0 {
 		return Record{}, ErrNotFound
 	}
 
 	return Record{
-		Country:    country,
-		Location:   locodeCSV.locationName,
-		SubDivName: locodeCSV.subDivName,
-		SubDivCode: locodeCSV.subDivCode,
-		Point:      locodeCSV.point,
-		Cont:       locodeCSV.continent,
+		Country:    cd.name,
+		Location:   locFromCSV(&cd.locodes[n]),
+		SubDivName: divNameFromCSV(&cd.locodes[n]),
+		SubDivCode: divCodeFromCSV(&cd.locodes[n]),
+		Point:      cd.locodes[n].point,
+		Cont:       cd.locodes[n].continent,
 	}, nil
 }
+
+func codeFromCSV(c *locodesCSV) string {
+	return string(c.code[:])
+}
+
+func locFromCSV(c *locodesCSV) string {
+	return locodeStrings[int(c.offset) : int(c.offset)+int(c.locationLen)]
```

Comment:
casts may be reduced
```suggestion
	return locodeStrings[c.offset : c.offset+uint32(c.locationLen)]
```

same for other funcs